libdevcrypto/SecretStore.cpp

```diff
@@ -132,50 +136,53 @@ void SecretStore::clearCache() const
 	m_cached.clear();
 }
 
-void SecretStore::save(std::string const& _keysPath)
+void SecretStore::save(string const& _keysPath)
 {
+	if (_keysPath.empty())
+		return;
+
 	fs::path p(_keysPath);
-	boost::filesystem::create_directories(p);
+	fs::create_directories(p);
 	for (auto& k: m_keys)
 	{
-		std::string uuid = toUUID(k.first);
-		std::string filename = (p / uuid).string() + ".json";
+		string uuid = toUUID(k.first);
+		string filename = (p / uuid).string() + ".json";
 		js::mObject v;
 		js::mValue crypto;
-		js::read_string(k.second.first, crypto);
+		js::read_string(k.second.encryptedKey, crypto);
 		v["crypto"] = crypto;
 		v["id"] = uuid;
 		v["version"] = c_keyFileVersion;
 		writeFile(filename, js::write_string(js::mValue(v), true));
-		if (!k.second.second.empty() && k.second.second != filename)
-			boost::filesystem::remove(k.second.second);
-		k.second.second = filename;
+		swap(k.second.filename, filename);
+		if (!filename.empty() && !fs::equivalent(filename, k.second.filename))
+			fs::remove(filename);
```

Comment:
Suggestion for future: boost vs write && rename on success. Some file systems act differently when fs errors occur (disk space, permissions, etc.). And unfortunately, due to default partitioning schemes, dual-boot scenarios, etc., running out of disk space is something which happens a *LOT* (among thousands of users). Doing something similar to vim would save someone in both of these situations: write-out _file.json, delete file.json, then rename _file.json to file.json. If something bad happens in the middle it's recoverable.